packages/php-wasm/web/vite.config.ts

```diff
@@ -59,93 +61,23 @@ export default defineConfig(({ command }) => {
 					}
 				},
 			},
-			/**
-			 * Vite can't extract static asset in the library mode:
-			 * https://github.com/vitejs/vite/issues/3295
-			 *
-			 * This workaround replaces the actual php_5_6.js modules paths used
-			 * in the dev mode with their filenames. Then, the filenames are marked
-			 * as external further down in this config. As a result, the final
-			 * bundle contains literal `import('php_5_6.js')` and
-			 * `import('php_5_6.wasm')` statements which allows the consumers to use
-			 * their own loaders.
-			 *
-			 * This keeps the dev mode working AND avoids inlining 5mb of
-			 * wasm via base64 in the final bundle.
-			 */
-			{
-				name: 'preserve-php-loaders-imports',
-
-				resolveDynamicImport(specifier): string | void {
-					if (
-						command === 'build' &&
-						typeof specifier === 'string' &&
-						specifier.match(/php_\d_\d\.js$/)
-					) {
-						/**
-						 * The ../ is weird but necessary to make the final build say
-						 * import("./php/jspi/php_8_2.js")
-						 * and not
-						 * import("php/jspi/php_8_2.js")
-						 *
-						 * The slice(-3) will ensure the 'php/jspi/'
-						 * portion of the path is preserved.
-						 */
-						return '../' + specifier.split('/').slice(-3).join('/');
-					}
+			vitePreserveLoadersImports([
+				{
+					regex: /php_\d_\d\.js$/,
+					transform: (specifier) =>
+						`../${specifier.split('/').slice(-3).join('/')}`,
```

Comment:
It's not easy to parse just by looking at it, let's document these with examples of paths we're targeting with this expression